### PR TITLE
Add auto-generated plugin reference to the `2.x` site

### PR DESCRIPTION
**Pull request by vy**
This PR adds the following new steps to the project build.

1. At `compile` phase, auto-generate plugin descriptors using an annotation processor
2. At `site` phase, using the generated plugin descriptors, auto-generate
   1. `/target/log4j-config.xsd`
   2. `/target/generated-source/site/plugin-descriptors.adoc` et al.

Employed annotation processor, XSD and AsciiDoc generators are provided by the Log4j Docgen shipped with the recently released Log4j Tools `0.8.0`.

This work is a part of #1954.

### CODE REVIEW

File: log4j-api/pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
   <name>Apache Log4j API</name>
   <description>The Apache Log4j API</description>
   <properties>
-    <javadoc.skip>false</javadoc.skip>
+    <maven.javadoc.skip>false</maven.javadoc.skip>
```
Comment (ppkarwasz) [inline log4j-api/pom.xml:31]: This PR also replaces our custom `javadoc.skip` property with the official one.

Comment (vy) [reply log4j-api/pom.xml:31]: Yes, and this is in line with `main` too. But, what do you mean with your remark? Is this a problem and should be reverted? If so, why exactly?

File: log4j-core/pom.xml [hunk 2]
```diff
@@ -222,40 +233,61 @@
...
+                <!-- `org.apache.logging.log4j.docgen.processor.DescriptorGenerator` for generating `log4j-plugins.xml`: -->
                 <path>
-                  <groupId>com.google.errorprone</groupId>
-                  <artifactId>error_prone_core</artifactId>
-                  <version>${error-prone.version}</version>
+                  <groupId>org.apache.logging.log4j</groupId>
+                  <artifactId>log4j-docgen</artifactId>
+                  <version>${log4j-docgen.version}</version>
                 </path>
               </annotationProcessorPaths>
```
Comment (ppkarwasz) [inline log4j-core/pom.xml:261]: It makes more sense to me to use the `process-plugins` execution for both the `PluginProcessor` and `DescriptorGenerator` processors.

File: log4j-core/src/main/java/org/apache/logging/log4j/core/config/xml/XmlConfiguration.java [hunk 3]
```diff
@@ -142,7 +141,7 @@ public XmlConfiguration(final LoggerContext loggerContext, final ConfigurationSo
             try (final InputStream is =
                     Loader.getResourceAsStream(schemaResource, XmlConfiguration.class.getClassLoader())) {
                 if (is != null) {
-                    final javax.xml.transform.Source src = new StreamSource(is, LOG4J_XSD);
+                    final javax.xml.transform.Source src = new StreamSource(is);
```
Comment (ppkarwasz) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/config/xml/XmlConfiguration.java:144]:
```suggestion
                    final javax.xml.transform.Source src = new StreamSource(is, schemaResource);
```

The `systemId` parameter is very useful for debugging (it is included in the exception messages).

**Remark**: this code is only executed when using the [strict XML configuration mode](https://logging.apache.org/log4j/2.x/manual/configuration.html#strict-xml) that you PR makes completely obsolete. The XSD that you generate allows users to validate a "concise XML config" and does not have any support for the "strict XML config".

File: log4j-core-test/src/test/java/org/apache/logging/log4j/core/config/xml/XmlSchemaTest.java [hunk 4]
```diff
@@ -1,103 +0,0 @@
...
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.MethodSource;
-import org.xml.sax.Attributes;
-import org.xml.sax.InputSource;
-import org.xml.sax.SAXException;
-import org.xml.sax.SAXParseException;
-import org.xml.sax.helpers.XMLFilterImpl;
-import org.xml.sax.helpers.XMLReaderFactory;
-
-public class XmlSchemaTest {
```
Comment (ppkarwasz) [inline log4j-core-test/src/test/java/org/apache/logging/log4j/core/config/xml/XmlSchemaTest.java:43]: Can you add a "feature request" issue to validate the configuration files in `log4j-core-test` using the generated schema?

**Review by ppkarwasz — APPROVED** [review]
Looks good to me.

File: log4j-core-java9/pom.xml [hunk 5]
```diff
@@ -30,6 +30,8 @@
   <properties>
     <maven.compiler.release>9</maven.compiler.release>
     <maven.deploy.skip>true</maven.deploy.skip>
+    <maven.install.skip>true</maven.install.skip>
```
[the remark this answers is not in the document]
Comment (vy) [reply log4j-core-java9/pom.xml]: Made it `./mvnw install`able again in a4851854936aea9ebc862b12e4ba0025abd74c47.

File: log4j-core/pom.xml (hunk 2 above)
Comment (vy) [reply log4j-core/pom.xml]: Fixed in 5aa7a3d16cf7c7b9ba4c9d15be66914da21e6e23.

File: log4j-core-test/src/test/java/org/apache/logging/log4j/core/config/xml/XmlSchemaTest.java (hunk 4 above)
Comment (vy) [reply log4j-core-test/src/test/java/org/apache/logging/log4j/core/config/xml/XmlSchemaTest.java:43]: Created #2406.

File: log4j-api/pom.xml (hunk 1 above)
Comment (ppkarwasz) [reply log4j-api/pom.xml:31]: No, it is just a note for future readers if Git blame points them to this PR.

**Pull request merged**